### PR TITLE
standardize ordering of names in reducer

### PR DESCRIPTION
**Pull request by bakkot**

### CODE REVIEW

File: src/main/java/com/shapesecurity/shift/reducer/CloneReducer.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public Expression reduceCompoundAssignmentExpression(@NotNull CompoundAssignment
     @NotNull
     @Override
     public MemberExpression reduceComputedMemberExpression(@NotNull ComputedMemberExpression node, @NotNull Node object, @NotNull Node expression) {
-        return new ComputedMemberExpression((Expression) object, (ExpressionSuper) expression);
+        return new ComputedMemberExpression((Expression) expression, (ExpressionSuper) object);
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/shift/reducer/CloneReducer.java:134]:
```
ComputedMemberExpression((ExpressionSuper) object, (Expression) expression);
```

Comment (bakkot) [inline src/main/java/com/shapesecurity/shift/reducer/CloneReducer.java:134]: [Disagree](https://github.com/shapesecurity/shift-java/blob/d3c090ef2de9ba42a3f45fe0aa57aab1123827cc/generated-sources/main/java/com/shapesecurity/shift/ast/ComputedMemberExpression.java#L34). This was only working by accident (the actual parameters to `reduceComputedMemberExpression` here were `expression`, `object`).

Comment (michaelficarra) [inline src/main/java/com/shapesecurity/shift/reducer/CloneReducer.java:134]: But I thought you wanted the parameters in the order they appear in the grammar.

Comment (bakkot) [inline src/main/java/com/shapesecurity/shift/reducer/CloneReducer.java:134]: I agree that also standardizing the interface to the AST node constructors would also be valuable, but there's a lot less potential for confusion there, since those are typed. Also, I don't have the file which generated them.

Comment (michaelficarra) [inline src/main/java/com/shapesecurity/shift/reducer/CloneReducer.java:134]: Ah, I realise now that that change is not part of this PR.
